fabric-common/lib/DiscoveryHandler.js

```diff
@@ -365,11 +365,22 @@ class DiscoveryHandler extends ServiceHandler {
 		endorsement_plan.groups = {};
 		endorsement_plan.layouts = [{}]; // only one layout which will have all organizations
 
+		const notFound = [];
+
 		for (const mspid of required_orgs) {
 			logger.debug(`${method} - found org:${mspid}`);
-			endorsement_plan.groups[mspid] = {}; // make a group for each organization
-			endorsement_plan.groups[mspid].peers = JSON.parse(JSON.stringify(peers_by_org[mspid].peers)); // now put in all peers from that organization
-			endorsement_plan.layouts[0][mspid] = 1; // add this org to the one layout and require one peer to endorse
+			endorsement_plan.groups[mspid] = {}; // make a group for each
+			if (peers_by_org[mspid] && peers_by_org[mspid].peers) {
```

Comment:
I'm not sure if it's possible for `peers` to be present but empty. If it is, I guess this this also needs a `peers_by_org[mspid].peers.length > 0` test.

Reply:
good idea just in case